aviary/interface/methods_for_level2.py

```diff
@@ -2325,6 +2327,169 @@ def run_aviary_problem(self,
+        optimizer : str, optional
+            Optimizer to be used for solving the fallout mission.
+        max_iter : int, optional
+            Max number of iterations allowed for the optimizer.
+        verbosity : Verbosity or list, optional
+            If Verbosity.DEBUG, debug print options ['desvars','ln_cons','nl_cons','objs'] will be set.
+            If a list is provided, it will be used as the debug print options.
+        """
+        if phase_info is None:
+            phase_info = self.phase_info
+        if mission_mass is None:
+            mission_mass = self.get_val(Mission.Design.GROSS_MASS)
+        if payload_mass is None:
+            if self.mission_method is HEIGHT_ENERGY:
+                payload_mass = self.get_val(Aircraft.CrewPayload.TOTAL_PAYLOAD_MASS)
+            elif self.mission_method is TWO_DEGREES_OF_FREEDOM:
+                payload_mass = self.get_val(Aircraft.CrewPayload.PASSENGER_PAYLOAD_MASS)
+        if optimizer is None:
+            optimizer = self.driver.options["optimizer"]
+
+        design_range = self.get_val(Mission.Design.RANGE)
+
+        prob_fallout = _load_off_design(json_filename, ProblemType.FALLOUT, phase_info,
+                                        payload_mass, design_range, mission_mass)
+
+        prob_fallout.check_and_preprocess_inputs()
+        prob_fallout.add_pre_mission_systems()
+        prob_fallout.add_phases()
+        prob_fallout.add_post_mission_systems()
+        prob_fallout.link_phases()
+        prob_fallout.add_driver(optimizer, max_iter=max_iter, verbosity=verbosity)
+        prob_fallout.add_design_variables()
+        prob_fallout.add_objective()
+        prob_fallout.setup()
+        prob_fallout.set_initial_guesses()
+        prob_fallout.run_aviary_problem(record_filename='fallout_problem_history.db')
+        return prob_fallout
+
+    def save_sizing_to_json(self, json_filename='sizing_problem.json'):
+        """
+        This function saves an aviary problem object into a json file.
+
+        Parameters
+        ----------
+        aviary_problem: OpenMDAO Aviary Problem
+            Aviary problem object optimized for the aircraft design/sizing mission.
+            Assumed to contain aviary_inputs and Mission.Summary.GROSS_MASS
+        json_filename:   string
+            User specified name and relative path of json file to save the data into.
+        """
+
+        aviary_input_list = []
+        with open(json_filename, 'w') as jsonfile:
+            # Loop through aviary input datastructure and create a list
+            for data in self.aviary_inputs:
+                (name, (value, units)) = data
+                type_value = type(value)
+
+                # Get the gross mass value from the sizing problem and add it to input list
+                if name == Mission.Summary.GROSS_MASS or name == Mission.Design.GROSS_MASS:
+                    Mission_Summary_GROSS_MASS_val = self.get_val(
```

Comment:
So, the gross mass is the only variable we take from the model. Is there going to be support (in the future maybe) for more design variables in the design run? I think you might be able to curate a list from the design variables assigned in the original design problem, after removing any dynamic variables added for the trajectories.